client/src/pages/homepage.js

```diff
@@ -2,20 +2,108 @@ import React, { Component } from "react";
 import Navbar from "../components/navbar";
 import auth from "../authentication/auth-login";
 import { Redirect } from "react-router-dom";
+import { Container, Row, Col} from 'react-bootstrap';
+import axios from "axios";
 
 class HomePage extends Component {
-  state = {};
+  state = {
+    title: '',
+    body: '',
+    posts: []
+  };
+
+  userState = {
+    username: '',
+    userPosts: []
+  }
+
+  componentDidMount = () => {
+    this.getBlogPost();
+    this.getUserInformation();
+  };
+
+
+  getBlogPost = () => {
+    axios.get('http://localhost:8000/api')
+    .then((response) => {
+      const data = response.data;
+      this.setState ({ posts: data });
+      console.log('Data has been received!');
+    })
+    .catch(() => {
+      alert('Error retrieving data')
+    })
+  }
```

Comment:
These axios requests should probably have the JWT attached as a header (key value format being "token":sessionStorage.get('jwt')). It can be used client-side for authentication purposes and to know which specific user sent the request. See the follow/profile components, you can just copy paste it.

Reply:
Will do